### DISCUSSION TITLE
How to get current user?

### DISCUSSION THREAD

**Discussion by Anton-K-dev** [mesonbuild/meson#11781] [General]
Hello! Can anyone help how to get current user name on linux machine who runs meson build?
That's simple action that I can't resolve... I mean for example 
```
# Returns project name
meson.project_name() 
# How to get current user on linux machine?

```
Thanks in advance

**Comment by eli-schwartz** [comment]
What would be the overall goal of determining the login name? How would this be utilized in the build process?

**Reply by Anton-K-dev** [reply]
I just need to set on config data user variable as it is
```
conf_data = configuration_data()
conf_data.set('user', $USER)
```

**Reply by eli-schwartz** [reply]
But wouldn't that be wrong when a build bot produces redistributable binaries?

If you know the user is the same at runtime as it is at build time then can't you just look up the username at runtime rather than in the build system?

**Reply by Anton-K-dev** [reply]
If you can't help me with my question just don't waste your and my time. 
Simple example: to generate binary that can run only by user who had build it.

**Comment by Anton-K-dev** [comment]
Solution: 
`    res = run_command('sh', '-c', 'echo $USER', check: true)
    message('USER: ', res.stdout().strip())
`
